pandas/core/indexes/api.py

```diff
@@ -169,7 +169,10 @@ def _union_indexes(indexes, sort=True):
     if len(indexes) == 1:
         result = indexes[0]
         if isinstance(result, list):
-            result = Index(sorted(result))
+            if sort:
+                result = Index(sorted(result))
```

Comment:
make this simpler
```
if sort:
   result = sorted(result)
result = Index(result)
```

Reply:
Are you sure this change is required? I didn't think we needed to change anything in core for this and putting a breakpoint locally this wasn't hit when doing the following:

```python
>>> pd.DataFrame.from_dict({0: [1], 1: [2]}, dtype=None, orient="index")
```

which emulates the json construction

Reply:
What JSON tests fail if you don't make this change?

Reply:
Uh, I'm sorry. I think this is my mistake. 
If modify only _json.py and test files in the master branch, there is no error. Thank you.

I have a question because it's my first PR. I'd like to keep modifying this, but I'm afraid it's going to get messy.
Can I proceed with a new PR? Or should I just continue with the PR here?

Reply:
Continue with this PR. You should just be able to revert the changes on the tip of your local branch. You can probably just even `git checkout upstream/master pandas/core/indexes/api.py` and commit to push back up